### PR TITLE
[Ripple] Fix ripple layout issues with autolayout

### LINKED ISSUE #4533: [Cards] MDCCard may not correctly relayout its ink view layer when using Autolayout
From the internal bug:

> I'm using Autolayout to layout TableViewCells (was using CollectionViewCells before as well) and after several rotations of the device (and thus, size changes of the cells and the corresponding cards), the ink splash shown when tapping the card no longer fills the whole card (see attached video). The InkView's size matched the one of the card, but apparently the InkLayer has not been correctly laid out. I'm circumventing this issue by manually calling  setNeedsLayout and layoutIfNeeded on the InkView's layer in the TableViewCell's layoutSubviews method. This is easily reproducible on iOS 9, but I have also seen it on newer iOS versions.

---

This is an internal issue. If you are a Googler, please visit [b/111383478](http://b/111383478) for more details.

---
#### Internal data
- Associated internal bug: [b/111383478](http://b/111383478)

### LINKED ISSUE #4603: [Cards] Selected card's ink incorrect after rotation
Using the Edit/Reorder demo, switch to "Edit" mode and select a few cards while in portrait mode. Then rotate the simulator to landscape. Watch the ink on the selected cards centered incorrectly:

* Portrait
image

* Landscape
image

Note: I temporarily changed the ink color to 38% alpha so it is easier to see the ink:
cardCell.inkView.inkColor = [colorScheme.primaryColor colorWithAlphaComponent:0.38f];

---
#### Internal data
- Associated internal bug: [b/117177669](http://b/117177669)

### PR DESCRIPTION
**Pull request by yarneo**
This fixes a long standing issue in our Ink/Ripple implementation where with auto layout the ink view doesn't fit itself correctly when there is a re-layout, as an example during an orientation change.

There were a few things needed to be fixed/added to be able to make the Ripple work correctly:

1. every time MDCRippleView lays out its view, the MDCRippleLayer sublayers need to receive the new bounds and need to lay out the ripple based on the new bounds.
2. The MDCRippleView should at all times be the size of it's superView.
2. When an MDCRippleLayer exists and needs to be laid out again, it needs to recalculate the ripple layer's position and path. This can happen as an example during a ripple animation when an orientation change occurs. This can also happen when there is a ripple overlay that isn't transient is on a surface and there is an orientation change.
3. Lastly, moving the MDCRippleLayer animations away from `removeOnCompletion = NO;` to having the final visual values be set prior to the animation, solves an issue where a `setNeedsLayout` reverts certain animations (an animation over layer.position) back to their original values and causes the ripple to be incorrectly centered.

Resolves #4533 
Resolves #4603 

Below are provided gifs showing the fix.
Before:
beforeripplelayout

After:
afterripplelayout

### CODE REVIEW

File: components/Ripple/src/MDCRippleView.m [hunk 1]
```diff
@@ -60,6 +60,15 @@ - (void)commonMDCRippleViewInit {
 - (void)layoutSubviews {
   [super layoutSubviews];
   [self updateRippleStyle];
+  self.frame = self.superview.bounds;
```
Comment (romoore) [inline components/Ripple/src/MDCRippleView.m]: Please be sure to call `CGRectStandardize` on the superview's bounds.

File: components/Ripple/src/MDCRippleView.m [hunk 2]
```diff
@@ -60,6 +60,15 @@ - (void)commonMDCRippleViewInit {
...
+}
+
+- (void)layoutSublayersOfLayer:(CALayer *)layer {
+  [super layoutSublayersOfLayer:layer];
+  for (CALayer *sublayer in self.layer.sublayers) {
+    sublayer.frame = self.bounds;
```
Comment (romoore) [inline components/Ripple/src/MDCRippleView.m]: Call `CGRectStandardize` before assigning the `bounds` to the sublayer's `frame`.

**Review by romoore — APPROVED** [review]
Please try to create a snapshot test that captures the end result of the test case.  From the description, it seems to be:

1. Set the frame/bounds
1. Trigger the Ripple to start
1. Perform a bounds change
1. Take a snapshot of the view with updated Ripple

I may be off regarding the exact steps, but it seems like something that would be possible to capture in a snapshot test.

File: components/Ripple/src/MDCRippleView.m (hunk 1 above)
Comment (yarneo) [reply components/Ripple/src/MDCRippleView.m]: done

File: components/Ripple/src/MDCRippleView.m (hunk 2 above)
Comment (yarneo) [reply components/Ripple/src/MDCRippleView.m]: done

**Comment by yarneo** [comment]
Added snapshot tests to test the ripple view and how it reacts to a layout change

**Pull request merged**